### PR TITLE
Fail gracefully when item id can't be found

### PR DESCRIPTION
**Pull request by Ladsgroup**
Bug: https://phabricator.wikimedia.org/T164989

### CODE REVIEW

**Comment by Ladsgroup** [comment]
Also, Shouldn't this repo be moved to gerrit? or at least Wikidata or wmde org?

**Comment by Ladsgroup** [comment]
Travis failure doesn't seem related. We definitely need to move this to gerrit.

File: src/PropertySuggester/GetSuggestions.php [hunk 1]
```diff
@@ -88,12 +89,17 @@ public function execute() {
...
+			try {
+				$suggestions = $suggestionGenerator->generateSuggestionsByItem(
+					$params->entity,
+					$params->suggesterLimit,
+					$params->minProbability,
+					$params->context
+				);
+			} catch ( InvalidArgumentException $ex ) {
+				$id = $params->entity;
+				$this->dieWithError( "Invalid id: $id", 'no-such-entity', [ 'id' => $id ] );
```
Comment (thiemowmde) [inline src/PropertySuggester/GetSuggestions.php]: "Invalid ID" is a bit misleading. The same exception is thrown in both cases: invalid ID, and entity not found. I would reuse the existing `$ex->getMessage()` here.

**Comment by thiemowmde** [comment]
@Ladsgroup, try adding `"wikimedia/testing-access-wrapper": "~1.0"` to the `require-dev` section.

**Comment by Ladsgroup** [comment]
Done in https://github.com/Wikidata-lib/PropertySuggester/pull/202

File: src/PropertySuggester/GetSuggestions.php (hunk 1 above)
Comment (filbertkm) [inline src/PropertySuggester/GetSuggestions.php]: not sure this should this should dieWithError.

below, the suggestions (based on entity/properties) get merged with normal suggestions from wbsearchentities. maybe we still want to fall back to the normal suggestions in this case?

Comment (Ladsgroup) [inline src/PropertySuggester/GetSuggestions.php]: That's way beyond scope of this PR. I want this done to reduce logspam not to change PropertySuggestor behavior.

Comment (Ladsgroup) [inline src/PropertySuggester/GetSuggestions.php]: @thiemowmde Done

Comment (filbertkm) [inline src/PropertySuggester/GetSuggestions.php]: @Ladsgroup looked again at the bug and think the change is ok.

File: src/PropertySuggester/GetSuggestions.php [hunk 2]
```diff
@@ -88,12 +89,16 @@ public function execute() {
...
-			);
+			try {
+				$suggestions = $suggestionGenerator->generateSuggestionsByItem(
+					$params->entity,
+					$params->suggesterLimit,
+					$params->minProbability,
+					$params->context
+				);
+			} catch ( InvalidArgumentException $ex ) {
+				$this->dieWithError( $ex->getMessage() );
```
Comment (thiemowmde) [inline src/PropertySuggester/GetSuggestions.php]: I think this should be `dieWithException`.

**Comment by Ladsgroup** [comment]
Done

**Comment by thiemowmde** [comment]
Did you forgot to push? I don't see a change.

**Comment by Ladsgroup** [comment]
Done

**Pull request merged**